src/i965_encoder_vp8.c

```diff
@@ -0,0 +1,6239 @@
+
+    // Scoreboard 1
+    gpe_context->vfe_desc6.scoreboard1.delta_x1 = 0x0;
+    gpe_context->vfe_desc6.scoreboard1.delta_y1 = 0xF;
+
+    // Scoreboard 2
+    gpe_context->vfe_desc6.scoreboard1.delta_x2 = 0x1;
+    gpe_context->vfe_desc6.scoreboard1.delta_y2 = 0xF;
+
+    // Scoreboard 3
+    gpe_context->vfe_desc6.scoreboard1.delta_x3 = 0xF;
+    gpe_context->vfe_desc6.scoreboard1.delta_y3 = 0xF;
+
+    // Scoreboard 4
+    gpe_context->vfe_desc7.scoreboard2.delta_x4 = 0xF;
+    gpe_context->vfe_desc7.scoreboard2.delta_y4 = 0x1;
+
+    // Scoreboard 5
+    gpe_context->vfe_desc7.scoreboard2.delta_x5 = 0x0;
+    gpe_context->vfe_desc7.scoreboard2.delta_y5 = 0xE;
+    // Scoreboard 6
+    gpe_context->vfe_desc7.scoreboard2.delta_x6 = 0x1;
+    gpe_context->vfe_desc7.scoreboard2.delta_y6 = 0xE;
+    // Scoreboard 7
+    gpe_context->vfe_desc7.scoreboard2.delta_x6 = 0xF;
+    gpe_context->vfe_desc7.scoreboard2.delta_y6 = 0xE;
+}
+
+static void
+i965_add_2d_gpe_surface(VADriverContextP ctx,
+                        struct intel_encoder_context *encoder_context,
+                        struct i965_gpe_context *gpe_context,
+                        struct object_surface *obj_surface,
+                        int is_uv_surface,
+                        int is_media_block_rw,
+                        unsigned int format,
+                        int index)
+{
+    struct i965_encoder_vp8_context *vp8_context = encoder_context->vme_context;
+    struct i965_gpe_table *gpe = vp8_context->gpe_table;
+    struct i965_gpe_resource gpe_resource;
+    struct i965_gpe_surface gpe_surface;
+
+    memset(&gpe_surface, 0, sizeof(gpe_surface));
+
+    i965_object_surface_to_2d_gpe_resource(&gpe_resource, obj_surface);
+    gpe_surface.gpe_resource = &gpe_resource;
+    gpe_surface.is_2d_surface = 1;
+    gpe_surface.is_uv_surface = !!is_uv_surface;
+    gpe_surface.is_media_block_rw = !!is_media_block_rw;
+
+    gpe_surface.cacheability_control = vp8_context->mocs;
+    gpe_surface.format = format;
+
+    gpe->context_add_surface(gpe_context, &gpe_surface, index);
+    i965_free_gpe_resource(&gpe_resource);
+}
+
+static void
+i965_add_adv_gpe_surface(VADriverContextP ctx,
+                         struct intel_encoder_context *encoder_context,
+                         struct i965_gpe_context *gpe_context,
+                         struct object_surface *obj_surface,
+                         int index)
+{
+    struct i965_encoder_vp8_context *vp8_context = encoder_context->vme_context;
+    struct i965_gpe_table *gpe = vp8_context->gpe_table;
+    struct i965_gpe_resource gpe_resource;
+    struct i965_gpe_surface gpe_surface;
+
+    memset(&gpe_surface, 0, sizeof(gpe_surface));
+
+    i965_object_surface_to_2d_gpe_resource(&gpe_resource, obj_surface);
+    gpe_surface.gpe_resource = &gpe_resource;
+    gpe_surface.is_adv_surface = 1;
+    gpe_surface.cacheability_control = vp8_context->mocs;
+    gpe_surface.v_direction = I965_VDIR
```

Comment:
Like as VDEnc, we will use multiple but small batchbuffers for multi-pass scenarios